[Bug][Sort] Icebreg connector has null pointer exception

### What happened

The null pointer exception occurred in the all database migration of icebreg connector.

```
aused by: org.apache.flink.client.program.ProgramInvocationException: The main method caused an error: null
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:117) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.client.deployment.application.ApplicationDispatcherBootstrap.runApplicationEntryPoint(ApplicationDispatcherBootstrap.java:245) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
	... 10 more
Caused by: java.lang.NullPointerException
	at org.apache.inlong.sort.iceberg.sink.FlinkSink$Builder.appendMultipleCommitter(FlinkSink.java:544) ~[pool-1-thread-7-1676344725852-test-sort-connector-iceberg-dlc-1.5.0-SNAPSHOT.jar:1.4.0]
	at org.apache.inlong.sort.iceberg.sink.FlinkSink$Builder.chainIcebergMultipleSinkOperators(FlinkSink.java:451) ~[pool-1-thread-7-1676344725852-test-sort-connector-iceberg-dlc-1.5.0-SNAPSHOT.jar:1.4.0]
	at org.apache.inlong.sort.iceberg.sink.FlinkSink$Builder.chainIcebergOperators(FlinkSink.java:395) ~[pool-1-thread-7-1676344725852-test-sort-connector-iceberg-dlc-1.5.0-SNAPSHOT.jar:1.4.0]
	at org.apache.inlong.sort.iceberg.sink.FlinkSink$Builder.append(FlinkSink.java:476) ~[pool-1-thread-7-1676344725852-test-sort-connector-iceberg-dlc-1.5.0-SNAPSHOT.jar:1.4.0]
	at org.apache.inlong.sort.iceberg.IcebergTableSink.lambda$getSinkRuntimeProvider$0(IcebergTableSink.java:129) ~[pool-1-thread-7-1676344725852-test-sort-connector-iceberg-dlc-1.5.0-SNAPSHOT.jar:1.4.0]
	at org.apache.flink.table.planner.plan.nodes.exec.common.CommonExecSink.applySinkProvider(CommonExecSink.java:294) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.table.planner.plan.nodes.exec.common.CommonExecSink.createSinkTransformation(CommonExecSink.java:145) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.table.planner.plan.nodes.exec.stream.StreamExecSink.translateToPlanInternal(StreamExecSink.java:140) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.table.planner.plan.nodes.exec.ExecNodeBase.translateToPlan(ExecNodeBase.java:134) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.table.planner.delegation.StreamPlanner$$anonfun$1.apply(StreamPlanner.scala:70) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at org.apache.flink.table.planner.delegation.StreamPlanner$$anonfun$1.apply(StreamPlanner.scala:69) ~[flink-table-blink_2.11-1.13.6.jar:1.13.6]
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) ~[flink-dist_2.11-1.13.6.jar:1.13.6]
```

### What you expected to happen

Write iceberg succeeded.

### How to reproduce

Start a all database migration task to iceberg.

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/FlinkSink.java
+++ b/inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/FlinkSink.java
@@ -156,6 +156,7 @@ public class FlinkSink {
         private Table table;
         private TableSchema tableSchema;
         private ActionsProvider actionProvider;
+        private boolean overwrite = false;
         private boolean appendMode = false;
         private Integer writeParallelism = null;
         private List<String> equalityFieldColumns = null;
@@ -249,6 +250,7 @@ public class FlinkSink {
         }
 
         public Builder overwrite(boolean newOverwrite) {
+            this.overwrite = newOverwrite;
             writeOptions.put(FlinkWriteOptions.OVERWRITE_MODE.key(), Boolean.toString(newOverwrite));
             return this;
         }
@@ -525,8 +527,7 @@ public class FlinkSink {
         private SingleOutputStreamOperator<Void> appendMultipleCommitter(
                 SingleOutputStreamOperator<MultipleWriteResult> writerStream) {
             IcebergProcessOperator<MultipleWriteResult, Void> multipleFilesCommiter =
-                    new IcebergProcessOperator<>(new IcebergMultipleFilesCommiter(catalogLoader,
-                            flinkWriteConf.overwriteMode()));
+                    new IcebergProcessOperator<>(new IcebergMultipleFilesCommiter(catalogLoader, overwrite));
             SingleOutputStreamOperator<Void> committerStream = writerStream
                     .transform(operatorName(ICEBERG_MULTIPLE_FILES_COMMITTER_NAME), Types.VOID, multipleFilesCommiter)
                     .setParallelism(1)

